parsing/longident.mli

```diff
@@ -29,6 +29,9 @@ val flatten: t -> string list
 val unflatten: string list -> t option
 val last: t -> string
 val parse: string -> t
+(** This function may return syntactically invalid {!t}.
+    If you need precise parsing function , consider using one of the functions
+    in {!Parse.Longident}  *)
```

Comment:
I would suggest changing the name of this function to `parse_flat` or some other name that highlights the limitation. Otherwise it's just too tempting to use it. The very point of this PR is that downstream users should *not* use this function, so forcing them to think about what to use instead would be nice.

Reply:
It sounds like a good idea.